src/process.c

```diff
@@ -1186,6 +1193,33 @@ cc_oci_vm_launch (struct cc_oci_config *config)
 	 */
 	ret = cc_proxy_disconnect (config->proxy);
 
+	/* Before create pid file!
+	 *
+	 * Docker provides a cgroup path that MUST be created before pid file
+	 * workload pid MUST be copy to task and cgroup.procs notifying to docker
+	 * that the workload is part of a cgroup.
+	 * With this change docker WILL NOT create a new cgroup and WILL NOT copy
+	 * the workload pid to this new cgroup avoiding file descriptor leaks
+	 */
+	cgroup_dir = g_strdup_printf ("/sys/fs/cgroup/memory/%s",
+		config->oci.oci_linux.cgroupsPath);
+	g_mkdir_with_parents (cgroup_dir, CC_OCI_CGROUP_MODE);
+
+	buf_len = g_snprintf (buf, sizeof(buf), "%d", config->state.workload_pid);
+
+	cgroup_tasks = g_strdup_printf ("%s/tasks", cgroup_dir);
+	fd_tasks = open (cgroup_tasks, O_WRONLY);
+	if (write (fd_tasks, buf, (size_t)buf_len) <= 0) {
+		g_critical ("failed to copy workload pid to cgroup tasks");
```

Comment:
I'd personally check the return from `open` but if that fails the `write` will detect it. However, I'd add a call to `strerror(errno)`.